disco/state.py

```diff
@@ -285,6 +285,11 @@ def on_guild_member_add(self, event):
         if event.member.guild_id not in self.guilds:
             return
 
+        if (self.guilds[event.member.guild_id].member_count is not UNSET and
+                # Avoid adding duplicate events to member_count.
+                event.member.id not in self.guilds[event.member.guild_id].members):
```

Comment:
This should also never be the case, lets either remove this or log a warning for now.